backend/app/services/knowledge/knowledge_service.py

```
@@ -670,9 +798,8 @@ def update_knowledge_base_type(
 
         # Check permission for team knowledge base
         if kb.namespace != "default":
-            if not check_group_permission(
-                db, user_id, kb.namespace, GroupRole.Maintainer
-            ):
+            # Only Owner/Maintainer can update KB type (manage permission required)
+            if not _check_kb_permission(db, kb, user_id, "manage"):
                 raise ValueError(
                     "Only Owner or Maintainer can update knowledge base type in this group"
                 )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)